packages/mgt/src/graph/graph.user.ts

```diff
@@ -94,39 +105,39 @@ export async function getMe(graph: IGraph): Promise<User> {
  */
 
 export async function getUserWithPhoto(graph: IGraph, userId?: string): Promise<IDynamicPerson> {
-  const batch = graph.createBatch();
   let person = null as IDynamicPerson;
   let cache: CacheStore<CacheUser>;
+  // let userDetailsResponse = null;
```

Comment:
Good catch on this one. You are removing the batching for this and sending the photo and user requests one after the other which could slow down things. Think about how to optimize this with the batch while keeping the ability to cache. Potential implementation:

```js
let photo = getPhotoFromCache();
let user = getUserFromCache();

if (!photo && !user) {
    (photo, user) = getFromBatch();
    cachePhotoAndUser(photo, user);
} else if (!photo) {
    photo = getPhoto()
    cachePhoto(photo)
} else if (!user) {
    user = getUser();
    cacheUser(user);
}

return (photo, user);
```